Error messages from scalafmt should clearly be marked as such to not be confused with compiler error messages

If I add sbt-scalafmt with an empty .scalafmt.conf to https://github.com/scala/scala3-example-project and run `scalafmt` I get:
```scala
[info] Formatting 13 Scala sources...
[error] /home/smarter/opt/scala3-example-project/src/main/scala/IntersectionTypes.scala:4: error: ; expected but : found
[error] object IntersectionTypes:
[error]                         ^: /home/smarter/opt/scala3-example-project/src/main/scala/IntersectionTypes.scala
[error] (Compile / scalafmt) /home/smarter/opt/scala3-example-project/src/main/scala/IntersectionTypes.scala:4: error: ; expected but : found
[error] object IntersectionTypes:
[error]                         ^: /home/smarter/opt/scala3-example-project/src/main/scala/IntersectionTypes.scala
[error] Total time: 4 s, completed 4 mai 2021 16:19:19
```
Here it's clear that these errors come from scalafmt since that's the command I ran, but if I also set `scalafmtOnCompile := true` as some projects do, then these errors will show up in the output of `compile` and look like compiler errors. I've seen at least one person being confused by that (and I was also confused until I realized the error was probably coming from scalafmt). Therefore, it'd be nice if `scalafmt` was explicitly written down in each error message from the formatter.

--- a/plugin/src/main/scala/org/scalafmt/sbt/ScalafmtPlugin.scala
+++ b/plugin/src/main/scala/org/scalafmt/sbt/ScalafmtPlugin.scala
@@ -9,6 +9,7 @@ import sbt.librarymanagement.MavenRepository
 import sbt.util.CacheImplicits._
 import sbt.util.CacheStoreFactory
 import sbt.util.FileInfo
+import sbt.util.Level
 
 import scala.util.Failure
 import scala.util.Success
@@ -111,6 +112,13 @@ object ScalafmtPlugin extends AutoPlugin {
   private object FilterMode {
     val diffDirty = "diff-dirty"
     val diffRefPrefix = "diff-ref="
+  }
+
+  private class ScalafmtLogger(log: Logger) extends Logger {
+    override def trace(t: => Throwable): Unit = log.trace(t)
+    override def success(message: => String): Unit = success(message)
+    override def log(level: Level.Value, message: => String): Unit =
+      log.log(level, "scalafmt: " + message)
   }
 
   private class FormatSession(

--- a/plugin/src/main/scala/org/scalafmt/sbt/ScalafmtSbtReporter.scala
+++ b/plugin/src/main/scala/org/scalafmt/sbt/ScalafmtSbtReporter.scala
@@ -25,6 +25,7 @@ class ScalafmtSbtReporter(
   override def error(file: Path, message: String, e: Throwable): Unit = {
     def getMessage() = {
       val res = new StringWriter()
+      res.write("scalafmt: ")
       res.write(Option(message).getOrElse("failed"))
       res.write(" [")
       res.write(file.toString)

